extensions/azurecore/src/account-provider/azureAccountProviderService.ts

```diff
@@ -131,7 +131,10 @@ export class AzureAccountProviderService implements vscode.Disposable {
 			let tokenCacheKey = `azureTokenCache-${provider.metadata.id}`;
 			let simpleTokenCache = new SimpleTokenCache(tokenCacheKey, this._userStoragePath, noSystemKeychain, this._credentialProvider);
 			await simpleTokenCache.init();
-			let accountProvider = new AzureAccountProvider(provider.metadata as AzureAccountProviderMetadata, simpleTokenCache, this._context);
+
+			const isSaw: boolean = vscode.env.appName.toLowerCase().indexOf('saw') > 0;
```

Comment:
I feel like this should be a product config that's mixed in for saw builds instead - that'd be a better way of checking this IMO and would allow changing it without having to get into the code.

Reply:
Yep, that's planned but unfortunately we don't have a way of mixing in specific parts of package.json, it's all or nothing. 

And I definitely don't think its a good idea to keep our package.jsons in two places. We're planning on thinking this through a bit more and making these build time.

Reply:
Well, this would be in product.json. And really? It doesn't actually mixin the values (i.e. only write the ones that are in the mixin repo)? That's surprising to me.

Is this a discussion you and Anthony are having?

Reply:
Where in product.json would we place this? I was planning that eventually be able to merge various JSONs together and give the distro side the heavier weight in the merge.

This could've been changed with a simple build time config change.

And yeah, we eventually want to plan it properly and see what are the changes we need to make for build time to reduce stuff like this.

Reply:
Just add a new property I'd think.

Anyways I'll sign off since this isn't really blocking - although you could still at least use the product service and check the quality instead of getting the app name as that seems like a slightly better approach.

Reply:
Yeah I'm going to make a followup issue for that because what you suggest is also not where we want to go I think.